.github/workflows/goreleaser.yml

```diff
@@ -1,16 +1,166 @@
-name: GoReleaser
+name: Build and Release
 
 on:
   push:
-    tags:
+    tags:	
       - '*'
-      - '!latest'
 
+# There are two cases where this GitHub action will run:
+#
+# 1. We are releasing a new latest version for src-cli, as a major, minor, or patch
+#    update, for instance 4.0.2 -> 4.1.0, OR
+# 2. We are releasing a new minor/patch version for an older major/minor version of
+#    src-cli, for instance if the latest version is 4.0.2 and we're releasing 3.43.3
+#
+# In both cases, we want to run both goreleaser and npm to publish new versions in all the
+# places we care about. For goreleaser publishing to Homebrew, we need to publish to
+# different formulas depending on if we're case 1. or case 2, and neither goreleaser nor
+# Homebrew has a good way of handling this automatically. In the case of the former (a new
+# latest release), we must:
+#
+# 1. Copy the main formula in our Homebrew tap for the previous latest release to a
+#    versioned formula
+# 2. Build and publish the new release to the main formula
+# 3. Update the Homebrew symlink alias for the latest version
+#
+# In the case of the latter (a patch release for an older version), we only need to:
+#
+# 1. Build and publish the new release to a versioned Homebrew formula
+#
+# This action contains 5 jobs to accommodate both cases:
+#
+# release_type    [always]      - checks tags, determines if we're case 1. or case 2.
+# goreleaser_pre  [if case 1.]  - copies the main formula to a versioned formula
+# goreleaser      [always]      - runs tests, builds + publishes with goreleaser
+# goreleaser_post [if case 1.]  - updates symlink to latest version
+# npm             [always]      - publishes to npm
 jobs:
+  # release_type compares the most recently created tag to the highest versioned tag
+  # available on the repo to determine if this release is for a new latest version or a
+  # patch of an older version.
+  release_type:
+    name: Determine release type
+    runs-on: ubuntu-latest
+    outputs:
+      is_latest_version: ${{ env.is_latest_version }}
+      latest_tag: ${{ env.latest_tag }}
+      second_latest_tag: ${{ env.second_latest_tag }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Fetch most recent tag, latest (highest version) tag, and second latest tag
+        # For latest and second latest tags, we can't use git tag --sort=version:refname
+        # because git doesn't have a concept of pre-release versions and thus mis-sorts
+        # versions like 4.0.0-rc.0 *after* 4.0.0.
+        run: |
+          echo "most_recent_tag=$(git tag --sort=taggerdate | tail -1)" >> $GITHUB_ENV
+          echo "latest_tag=$(git tag | tr - \~ | sort --version-sort | tr \~ - | tail -1)" >> $GITHUB_ENV
+          echo "second_latest_tag=$(git tag | tr - \~ | sort --version-sort | tr \~ - | tail -2 | sed -n 1p)" >> $GITHUB_ENV
```

Comment:
As the comment above explains, `git tag --sort=version:refname` sorts by semantic version number but doesn't understand pre-release versions, so it sorts versions like `4.0.0-rc.0` after `4.0.0` when they should come before. I shamelessly stole this from a comment on [this gist](https://gist.github.com/loisaidasam/b1e6879f3deb495c22cc), but it worked reliably from my testing. To break it down, since I hate obtuse bash commands:

- `git tag` lists all tags in src-cli.
- `tr - \~` replaces hyphens with tildes, so `4.0.0-rc.0` becomes `4.0.0~rc.0` (this is apparently what's typically understood as pre-releases by certain Linux distros)
- `sort --version-sort` does natural sort by version number of the tags
- `tr \~ -` replaces the tildes with hyphens again
- `tail -1` takes the last item, which based on the sorting should be the latest version tag

The command for the second-latest tag is the same, except it takes the second-to-last item instead.